gremlin-dotnet/test/pom.xml

```diff
@@ -115,16 +116,29 @@ limitations under the License.
                                 </goals>
                                 <configuration>
                                     <skip>${skipTests}</skip>
-                                    <environmentVariables>
-                                        <GREMLIN_SERVER>${project.version}</GREMLIN_SERVER>
-                                        <HOME>${user.home}</HOME>
-                                    </environmentVariables>
+                                    <!-- don't need to set env variables for container tear down -->
                                     <executable>docker-compose</executable>
                                     <arguments>
                                         <argument>down</argument>
                                     </arguments>
                                 </configuration>
                             </execution>
+                            <execution>
+                                <id>remove-dangling-images</id>
+                                <phase>integration-test</phase>
+                                <goals>
+                                    <goal>exec</goal>
+                                </goals>
+                                <configuration>
+                                    <skip>${skipTests}</skip>
+                                    <executable>docker</executable>
+                                    <arguments>
+                                        <argument>image</argument>
+                                        <argument>prune</argument>
```

Comment:
So the [docker image prune](https://docs.docker.com/config/pruning/#prune-images) will only remove the dangling images on the machine. By definition they are old images that becomes untagged when you create a new version and no longer serve a purpose and consume disk space. 

Currently, we do the container removal with `docker compose down` and keep all tagged images on the machine.

Typically dangling images are safe to remove, but I will look into ways to remove only dangling images related to our build.

Reply:
I added a label filter option so that the dangling images we prune are only ones we created with the label "maintainer=dev@tinkerpop.apache.org" in [f92df86](https://github.com/apache/tinkerpop/pull/1788/commits/f92df86bb0533dfd9f95dfbe3f3b2875439b9af3). This should make sure we don't touch the other ones on the machine.